scenario/src/main/kotlin/com/cosmotech/scenario/azure/ScenarioServiceImpl.kt

```diff
@@ -723,14 +742,24 @@ internal class ScenarioServiceImpl(
     return true
   }
 
-  private fun updateScenarioOwner(existingScenario: Scenario, scenario: Scenario) {
-    // Allow to change the ownerId as well, but only the owner can transfer the ownership
-    if (existingScenario.ownerId != getCurrentAuthenticatedUserName()) {
-      // TODO Only the owner or an admin should be able to perform this operation
-      throw CsmAccessForbiddenException(
-          "You are not allowed to change the ownership of this Resource")
+  private fun updateScenarioOwner(
+      existingScenario: Scenario,
+      scenario: Scenario,
+      hasChanged: Boolean
+  ): Boolean {
+    return if (scenario.ownerId != null && scenario.changed(existingScenario) { ownerId }) {
+      // Allow to change the ownerId as well, but only the owner can transfer the ownership
+      if (existingScenario.ownerId != getCurrentAuthenticatedUserName()) {
+        // TODO Only the owner or an admin should be able to perform this operation
+        throw CsmAccessForbiddenException(
+            "You are not allowed to change the ownership of this Resource")
+      }
+      existingScenario.ownerId = scenario.ownerId
+
+      true
+    } else {
+      hasChanged
```

Comment:
I'm not really a huge fan of this '"if" syntax... could you please rewrite it in an old-school way :p ?